Stable cloud release merge into master from dev

dcelik opened the pull request.

jabb1123 committed:
Merge branch 'master' into dev

runnersaw committed:
Object Tracking runs on background thread

jabb1123 committed:
ADD:Functional subprocess error handling and optional email

runnersaw committed:
Make analysis handler run in background

runnersaw committed:
Fix analysis route; add create video thread

jabb1123 committed:
Merge branch 'master' into concurrency

jabb1123 committed:
Optimizing statusHandler with failure code

jabb1123 committed:
StatusHelper implemented in objectTracking

jabb1123 committed:
FIX: spelling, variable uses

jabb1123 committed:
Merge pull request #42 from santosfamilyfoundation/subprocess-error-handling

Functional subprocess error handling and optional email to analysis

jabb1123 committed:
Merge branch 'dev' into statusHandler

jabb1123 committed:
statusHelper implemented in all handlers where applicable

jabb1123 committed:
Merge branch 'dev' into statusHandler

# Conflicts:
#	app/handlers/objectTracking.py
#	app/handlers/safetyAnalysis.py
#	app/handlers/testConfig.py

jabb1123 committed:
Updated analysis to work with new threading

jabb1123 committed:
FIX: Naming from camel to snake

jabb1123 committed:
FIX: Another casing issue

jabb1123 committed:
FIX: Calling class via name

jabb1123 committed:
Casts vals to string for set and back when get

jabb1123 committed:
FIX: typo

runnersaw committed:
Test feature tracking works; object tracking in progress

jabb1123 committed:
Merge remote-tracking branch 'origin/concurrency' into concurrency

jabb1123 committed:
Rewrote `status.py` to work properly, removed int cast

jabb1123 committed:
ADD: StatusHelper initialized on project creation

runnersaw committed:
Test object tracking working

runnersaw committed:
Merge remote

jabb1123 committed:
Moved statusHelper to post in upload homography

jabb1123 committed:
Merge pull request #43 from santosfamilyfoundation/statusHandler

Status handler implemented in `TrafficCloud` for 5 handlers:

jabb1123 committed:
Merge branch 'dev' into concurrency

# Conflicts:
#	app/handlers/analysis.py
#	app/handlers/objectTracking.py
#	app/handlers/safetyAnalysis.py
#	app/handlers/testConfig.py

jabb1123 committed:
Merge branch 'dev' into concurrency

runnersaw committed:
Add video helper; Fix merge issues

runnersaw committed:
Fix some merge problems

runnersaw committed:
Test configuration working

Commit:
MOD: fix upload homography bug

Commit:
Merge branch 'dev' of https://github.com/santosfamilyfoundation/SantosCloud into dev

runnersaw committed:
Everything working except create highlight video

Commit:
MOD: Fixing import errors and references

Commit:
MOD: Fixing indentation

jabb1123 committed:
FIX: self. references

runnersaw committed:
Fix status helper

Commit:
Merge pull request #46 from santosfamilyfoundation/fix-status-helper

Fix status helper

runnersaw committed:
Base handler returns error properly

runnersaw committed:
Base handler returns error properly

runnersaw committed:
Merge dev

Commit:
ADD: make pdf report request handler (#45)

* ADD: make pdf report request handler

Summary:

- request handler takes all viz in final_images and puts them into a report called santosreport.pdf
- changed makePdf to take a out_path, so that the report could be saved in a different directory from final_images
- report saved as landscape letter size, with the assumption that the visualization images are landscape as well

* Added newline at EOF of makeReport.py

Commit:
Merge branch 'return-error' of https://github.com/santosfamilyfoundation/TrafficCloud into return-error

Conflicts:
	app/handlers/makeReport.py

runnersaw committed:
Fix highlight video

Commit:
Merge pull request #44 from santosfamilyfoundation/concurrency

Concurrency

runnersaw committed:
FIX: add error handling to make report route

runnersaw committed:
ADD: print errors

runnersaw committed:
Merge dev into return-error

Commit:
Merge pull request #47 from santosfamilyfoundation/return-error

Base handler returns error properly

Commit:
ADD: Matplotlib option

Commit:
Merge branch 'dev' of https://github.com/santosfamilyfoundation/SantosCloud into dev

Commit:
FIX: safety report opens in Preview on MacOS (#49)

Summary: Learned of an issue in pyfpdf where images that are
transparent (4 channel) cannot be viewed in certain viewers.
Now, we save all visualizations as 3 channel jpg images.

Tests: Generated new santosreport and was able to open in
Preview on MacOS.

Commit:
Merge branch 'dev' of https://github.com/santosfamilyfoundation/SantosCloud into dev

runnersaw committed:
Merge dev

runnersaw committed:
Edit video.py to delete all files, not just ones with extensions

runnersaw committed:
Remove images folder after completion

runnersaw committed:
Making videos works and removes temp images

runnersaw committed:
Fix

runnersaw committed:
FIX: delete_files function

runnersaw committed:
Remove print

jabb1123 committed:
Implemented enum in Status class

jabb1123 committed:
Merge branch 'dev' into StatusCheck

Commit:
Merge pull request #50 from santosfamilyfoundation/output-directory

Output directory

jabb1123 committed:
Fixed small errors. Tested and functional.

jabb1123 committed:
UPDATE: requirements.txt now reflects added enum package

jabb1123 committed:
FIX: indentation error

Commit:
FIX: Overlapping Plots of User Counts / SpeedCDF (#52)

Summary: Closing the user icon plot allows the other plot
to not be overwritten on the same figure.

runnersaw committed:
Also delete title slide images

Commit:
Merge pull request #53 from santosfamilyfoundation/delete-temp-video

Also delete title slide images

jabb1123 committed:
EDIT: to appease my robot overloads

jabb1123 committed:
ADD: More info on how to fix bad status

jabb1123 committed:
Testing precheck

segerphilip committed:
Retrieve results (#48)

* ADD: Begin zip process for file writeback

* MOD: stream the highlight video file

* EDIT: zip file and download untested

* EDIT: syntax

* EDIT: get body args method naming

* EDIT: I don't like atom

* EDIT: writing the right file now

* EDIT: fix for now, need to get zip working

* EDIT: file path naming

* EDIT: file path args

* EDIT: cleaned up for PR

* EDIT: better zip filepath, pr comments

* EDIT: error str and one letter identifier fix

* EDIT: file name and added the report result

* EDIT: remove zip after sending

jabb1123 committed:
Finished: precheck ensures no double calls on long functions

jabb1123 committed:
Refactor main tests to prepare. Exits code quicker

jabb1123 committed:
Fix: indentation

jabb1123 committed:
TestConfig working with pre-check

runnersaw committed:
Add default config handler

runnersaw committed:
Regenerate API documentation

jabb1123 committed:
FIX: All handlers now through error properly

runnersaw committed:
Returned values aren't optional

jabb1123 committed:
Merge pull request #51 from santosfamilyfoundation/StatusCheck

StatusCheck

jabb1123 committed:
ADD: test length limited as per API documentation.

Commit:
UPDATE: highlight video and speed distribution viz (#55)

Summary:

- highlight video will specify only the interacitng objects
- probability density function of speed is produced and saved in the report

Making the highlight video object numbers appear only for the objects of interest

runnersaw committed:
Changed to GET method

Commit:
Merge pull request #54 from santosfamilyfoundation/default-config

Add default config handler

Commit:
Removed uses of alterInteractionsTableWithRoadUser (#56)

* Removed uses of alterInteractionWithRoadUser

Summary: Per Nicolas's recommendation, altering the interactions
table was unncessary, since the same road user info could be got
from the objects table.  So removed all uses of alter interactions
table, since storage in TrafficIntelligence has always been purged
of it

* added output directory to display traj call

dcelik committed:
MOD: moving file download to inherited helper method (#57)

dcelik committed:
Homography fixes (#58)

* Cloud side fixes for homography upload

* MOD: Name change from uploadHomography to Config tomography

* MOD: apidocs updated to reflect change in uploadhomography

* Update configHomography.py

jabb1123 committed:
FIX: bugs in configHomography

segerphilip committed:
Retrieve test results (#59)

* ADD: get test config video start

* EDIT: check for report before retrieve results

* EDIT: error check for tests finished

* EDIT: changed to proper filepath

* EDIT: self identifier and test flag

dcelik committed:
Fixing status handler for analysis route + misc other status fixes (#60)

* MOD: consolidate statusHelper calls, add status checking to analysis route

* addressing fixes brought up in the PR

jabb1123 committed:
ADD: Status handler returns easier to parse dict

jabb1123 committed:
Fixed bad conversion

Commit:
Filename matches test type

Commit:
Rm speed CDF, replaced w/ speed Distribution (#63)

* Rm speed CDF, replaced w/ speed Distribution

Summary:

- We were currently making CDF / PDF distribution visualizations with
  each call of speedCDF.
- Now, we only make the PDF distribution
- Since CDF was removed, I've renamed it more aptly to "Distribution"
  so that the routes, handlers, and methods are no longer a misnomer

* Updated API docs for createSpeedDistribution and configHomography

runnersaw committed:
Cleanup IN_PROGRESS flags upon exit

runnersaw committed:
Cleanup on app start

Commit:
Merge pull request #65 from santosfamilyfoundation/cleanup-after-exit

Cleanup IN_PROGRESS flags upon exit

dcelik committed:
Fixes to checking and file creation (#64)

* Fixes to checking and file creation

* Remove print statements and bugs

segerphilip reviewed (approved):
I totally read it all 🚢 🇮🇹

Commit:
Highlight limit (#67)

* Sort and limit highlight video

* Update: highlight limit to match TI

Summary:
- Removed ordering by severity here, since its done by TI
- Added num_near_misses_to_use as an argument
- Set max limit to 10

runnersaw reviewed (approved):
:ship: :it:
